testutils/utils.go

```diff
@@ -0,0 +1,399 @@
+
+	consulServiceRouterController := &service.ConsulServiceRouteReconciler{
+		Client: k8sClient,
+		Log:    ctrl.Log.WithName("controllers").WithName("service").WithName("ConsulServiceRoute"),
+		Scheme: mgr.GetScheme(),
+	}
+
+	err = consulServiceRouterController.SetupWithManager(mgr)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	go func() {
+		err = mgr.Start(ctrl.SetupSignalHandler())
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+	}()
+}
+
+func CreateAndSetTestKubeconfig(hostWithPort string) error {
+	kubeconfig := apiv1.Config{
+		APIVersion:     "v1",
+		Kind:           "Config",
+		CurrentContext: "default",
+		Clusters:       []apiv1.NamedCluster{{Name: "default", Cluster: apiv1.Cluster{Server: fmt.Sprintf("http://%s", hostWithPort)}}},
+		Contexts:       []apiv1.NamedContext{{Name: "default", Context: apiv1.Context{Cluster: "default", Namespace: DefaultK8sNamespace, AuthInfo: "admin"}}},
+		AuthInfos:      []apiv1.NamedAuthInfo{{Name: "admin", AuthInfo: apiv1.AuthInfo{Username: "admin", Password: ""}}},
+	}
+
+	serializedConfig, err := json.Marshal(kubeconfig)
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(kubeconfigPath, serializedConfig, 0644)
+	if err != nil {
+		return nil
+	}
+
+	err = os.Setenv("KUBECONFIG", kubeconfigPath)
+
+	return err
+}
+
+func StartConsulLocalEnv() error {
+	if consulCmd == nil {
+		logf.Log.Info("starting local consul")
+
+		consulCmd = exec.Command(path.Join(testBinDir, "consul"), "agent", "-dev")
+		if showConsulLogs {
+			consulCmd.Stdout = os.Stdout
+			consulCmd.Stderr = os.Stderr
+		}
+
+		err := consulCmd.Start()
+		if err != nil {
+			return err
+		}
+
+		time.Sleep(time.Second)
+
+		logf.Log.Info("started local consul")
+	}
+
+	if consulK8sCmd == nil {
+		logf.Log.Info("creating consul CRDs in the current k8s cluster")
+		err := exec.Command("make", "setup-consul-test-env", "-f", os.Getenv("MAKEFILE_PATH")).Run()
+		if err != nil {
+			return err
+		}
+
+		logf.Log.Info("created consul CRDs in the current k8s cluster")
+
+		logf.Log.Info("starting local consul-k8s")
+
+		consulK8sCmd = exec.Command(path.Join(testBinDir, "consul-k8s"), "controller", "-enable-webhooks=false", "-datacenter", "dc1")
+		if showConsulLogs {
+			consulK8sCmd.Stdout = os.Stdout
+			consulK8sCmd.Stderr = os.Stderr
+		}
+
+		err = consulK8sCmd.Start()
+		if err != nil {
+			return err
+		}
+
+		time.Sleep(time.Second)
+
+		logf.Log.Info("started local consul-k8s")
+	}
+
+	return nil
+}
+
+func StopConsulLocalEnv() error {
+	var consulStopErr error
+	var consulK8sStopErr error
+	if consulCmd != nil {
+		consulStopErr = consulCmd.Process.Kill()
+	}
+
+	if consulK8sCmd != nil {
+		consulK8sStopErr = consulK8sCmd.Process.Kill()
+	}
+
+	if consulStopErr != nil {
+		return consulStopErr
+	}
+
+	if consulK8sStopErr != nil {
+		return consulK8sStopErr
+	}
+
+	return nil
+}
+
+func CreateServiceDefaults(ctx context.Context, k8sClient client.Client, service string, protocol ...string) error {
```

Comment:
IMO having a `protocol` rest parameter here is gratuitous because you're never using it.
I don't think there's a point in leaving something so uncertain for the future generations